### PR TITLE
API handling for required contact info

### PR DESCRIPTION
**Pull request by Omarkojak**
for issue #162

### CODE REVIEW

File: app/Http/Controllers/API/EventAPIController.php [hunk 1]
```diff
@@ -103,8 +103,10 @@ public function unfollow(Request $request, $id)
 
     public function register(Request $request, $id)
     {
-        $this->eventService->register($id, $request->get('volunteer'));
-        return response()->json(['message' => 'Success.'], 200);
+        $validator = $this->eventService->register($id, $request->get('volunteer'));
+        if($validator->passes())
+            return response()->json(['message' => 'Success.'], 200);
+        return response()->json(['message' => 'Registration Failed.', 'errors' => $validator->errors()], 200);
```
Comment (AhmadElsagheer) [inline app/Http/Controllers/API/EventAPIController.php]: status code = 400 (bad request)

Comment (Omarkojak) [inline app/Http/Controllers/API/EventAPIController.php]: dh eli hwa f errors ? line 109

Comment (AhmadElsagheer) [inline app/Http/Controllers/API/EventAPIController.php]: yes. don't return 200 OK for a failed request

**Comment by Omarkojak** [comment]
@AhmadElsagheer

**Pull request merged**